src/client/components/ProductsPagination/ProductsPagination.component.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import './ProductsPagination.styles.css';
+
+const ProductsPagination = ({ productsImages }) => {
+  const [productsPerPage] = useState(2);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const [pageNumberLimit, setPageNumberLimit] = useState(5);
+  const [maxPageNumberLimit, setMaxPageNumberLimit] = useState(5);
+  const [minPageNumberLimit, setMinPageNumberLimit] = useState(0);
+
+  const pages = [];
+  for (
+    let i = 1;
+    i <= Math.ceil(productsImages.length / productsPerPage);
+    i += 1
+  ) {
+    pages.push(i);
+  }
+
+  const indexOfLastProduct = currentPage * productsPerPage;
+  const indexOfFirstProduct = indexOfLastProduct - productsPerPage;
+  const currentProducts = productsImages.slice(
+    indexOfFirstProduct,
+    indexOfLastProduct,
+  );
+
+  const paginate = (pageNumber) => setCurrentPage(pageNumber);
+
+  const renderPageNumbers = pages.map((number) => {
+    if (number < maxPageNumberLimit + 1 && number > minPageNumberLimit) {
+      return (
+        <li
+          key={number}
+          onClick={() => paginate(number)}
+          className={currentPage === number ? 'active' : null}
+        >
+          {number}
+        </li>
+      );
+    }
+    return null;
+  });
+
+  const displayProducts =
+    currentProducts &&
+    currentProducts.map((image) => {
+      return (
+        <div>
+          <img src={image} alt="product-img" height="300px" width="280px" />
+        </div>
+      );
+    });
+
+  const handleNextBtn = () => {
+    setCurrentPage(currentPage + 1);
```

Comment:
Checking the currentPage value before setting the state makes more sense IMO.
If currentPage === maxPageNumber don't increment.